lib/RoadizCoreBundle/src/Entity/Node.php

```diff
@@ -710,8 +712,13 @@ public function getStackTypes(): Collection
      */
     public function addStackType(NodeType $stackType): static
     {
-        if (!$this->getStackTypes()->contains($stackType)) {
-            $this->getStackTypes()->add($stackType);
+        if (!$this->getStackTypes()->contains($stackType->getName())) {
```

Comment:
I believe you confused yourself here because of $stackType var name. But it is a NodeType here :

StackTypes collection contains a collection of StackType object.
So it will never contain a `string`.  `$this->getStackTypes()->contains($stackType->getName())` will always be false!

Use $this->getStackTypes()->exists method instead with a callable which compare StackType::getName against your NodeType name